packages/runtimes/js/src/JSRuntime.js

```diff
@@ -570,13 +587,35 @@ function getRelativePathExpr(
   to: NamedBundle,
   options: PluginOptions,
 ): string {
+  let relativePath = relativeBundlePath(from, to, {leadingDotSlash: false});
   if (shouldUseRuntimeManifest(from, options)) {
-    return `require('./relative-path')(${JSON.stringify(
-      from.publicId,
-    )}, ${JSON.stringify(to.publicId)})`;
+    // Get the relative part of the path. This part is not in the manifest, only the basename is.
+    let relativeBase = path.posix.dirname(relativePath);
+    if (relativeBase === '.') {
+      relativeBase = '';
+    } else {
+      relativeBase = `${JSON.stringify(relativeBase + '/')} + `;
+    }
+    return (
+      relativeBase +
+      `require('./bundle-manifest').resolve(${JSON.stringify(to.publicId)})`
+    );
   }
 
-  return JSON.stringify(relativeBundlePath(from, to, {leadingDotSlash: false}));
+  return JSON.stringify(relativePath);
+}
+
+function getAbsoluteUrlExpr(relativePathExpr: string, bundle: Bundle) {
+  if (
+    bundle.env.outputFormat === 'esmodule' &&
+    (bundle.env.isLibrary || bundle.env.supports('import-meta-url'))
+  ) {
+    return `new __parcel__URL__(${relativePathExpr}, import.meta.url).toString()`;
```

Comment:
For library builds, this might not fully fix the problem because the `new URL` call ends up in a separate module with the worker doing something like `new Worker(parcelRequire('...'))`. That's not gonna be statically analyzable by another bundler. Not really sure how to fix this though. Runtimes don't modify the compiled code of the original module...